src/mtype.c

```diff
@@ -3980,6 +3980,8 @@ void TypeSArray::resolve(Loc loc, Scope *sc, Expression **pe, Type **pt, Dsymbol
     }
     else
     {
+        if ((*pt)->ty != Terror)
+            next = *pt;     // prevent re-running semantic() on 'next'
```

Comment:
This is a bit hackish. It relies on the presumption that TypeSArray::resolve() is only called with `next` as the `this` pointer, making the logic very hard to follow.